sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -394,6 +394,15 @@ class DataFrame private[sql](
   def show(numRows: Int, truncate: Boolean): Unit = println(showString(numRows, truncate))
   // scalastyle:on println
 
+  def show(numRows: Int, truncate: Boolean, outputStream: OutputStream): Unit = {
+    val writer = new OutputStreamWriter(outputStream, "UTF-8")
+    writer.write(showString(numRows, truncate))
+  }
+
+  def show(numRows: Int, truncate: Boolean, writer: Writer): Unit = {
```

Comment:
these signatures are really clunky and I don't think we want it. Also the string is already composed so it's not like we're streaming things to the writer anyway.